presto-main/src/main/java/com/facebook/presto/operator/scalar/MapDistinctFromOperator.java

```diff
@@ -72,11 +72,8 @@ public static boolean isNotDistinctFrom(
     {
         boolean leftNull = leftBlock == null;
         boolean rightNull = rightBlock == null;
-        if (leftNull != rightNull) {
-            return false;
-        }
-        if (leftNull) {
-            return true;
+        if (leftNull || rightNull) {
+            return leftNull & rightNull;
```

Comment:
I believe `&&` is more proper here than `&` because we are not trying to do a bitwise operation here.